public/js/node-history-plot.js

```diff
@@ -0,0 +1,90 @@
+(function() {
+
+  let svg = d3.select('#node-history-plot-svg');
+  let svgWidth = svg.node().getBoundingClientRect().width,
+      svgHeight = svg.node().getBoundingClientRect().height;
+
+  let plotMargin = 40;
+  let plotWidth = svgWidth - 2 * plotMargin,
+      plotHeight = svgHeight - 2 * plotMargin;
+
+  svg.style('margin-left', `-${plotMargin + 5}px`);
+
+  init();
+
+  function init() {
+    // This is the entry point. Get route table from API,
+    // convert to list of nodes and links, then render.
+    fetch('/api/v0/numNodesTimeseries')
+      .then((response) => response.json())
+      .then(({ numGateways, numNodes, timestamps }) => render(numGateways, numNodes, timestamps));
+  }
+
+  function render(numGateways, numNodes, timestamps) {
+    timestamps = timestamps.map((t) => new Date(t));
+    
+    let xScale = window.xScale = d3.scaleTime()
+      .domain(d3.extent(timestamps))
+      .range([0, plotWidth]);
+    let yScale = window.yScale = d3.scaleLinear()
+      .domain([0, d3.max(Array.prototype.concat(numGateways, numNodes))])
+      .range([plotHeight, 0]);
+    let mainGroup = svg.append('g')
+      .attr('transform', `translate(${plotMargin}, ${plotMargin})`);
+
+    let nodeData = _.zip(timestamps, numNodes).map((el) => {
+      return {
+        timestamp: el[0],
+        value: el[1]
+      };
+    });
+
+    let gatewayData = _.zip(timestamps, numGateways).map((el) => {
+      return {
+        timestamp: el[0],
+        value: el[1]
+      };
+    });
+
+    console.log(nodeData, gatewayData);
```

Comment:
Consider using `console.debug` instead to not always show debug output in prod unless a developer wants to see it.

Reply:
deleted